### PR TITLE
mimir worker v0

### PR DESCRIPTION
**Pull request by PietroPasotti**
This PR adds the `mimir_worker` interface.

A follow-up PR will align with the contents of the mimir charm repo and add some tests

### CODE REVIEW

**Review by simskij — APPROVED** [review]
Merging with the caveat that this remains `Draft` until we settle on the final format for v0.

File: README.md [hunk 1]
```diff
@@ -26,6 +26,7 @@ To quickly get started, see the [template interface](https://github.com/canonica
 | Observability | [`grafana_auth`](interfaces/grafana_auth/v0/README.md)                       | ![Status: Draft](https://img.shields.io/badge/Status-Draft-orange)  |
 |               | [`ingress`](interfaces/ingress/v0/README.md)                                 | ![Status: Live](https://img.shields.io/badge/Status-Live-darkgreen) |
 |               | [`ingress_per_unit`](interfaces/ingress_per_unit/v0/README.md)               | ![Status: Live](https://img.shields.io/badge/Status-Live-darkgreen) |
+|               | [`mimir_worker`](interfaces/mimir_worker/v0/README.md)             | ![Status: Live](https://img.shields.io/badge/Status-Draft-orange) |
```
Comment (gruyaume) [inline README.md:29]:
```suggestion
|               | [`mimir_worker`](interfaces/mimir_worker/v0/README.md)             | ![Status: Draft](https://img.shields.io/badge/Status-Draft-orange) |
```

File: docs/json_schemas/mimir_worker/v0/provider.json [hunk 2]
```diff
@@ -0,0 +1,51 @@
+{
+  "title": "ProviderSchema",
+  "description": "Provider schema for Ingress.",
```
Comment (gruyaume) [inline docs/json_schemas/mimir_worker/v0/provider.json:3]: Ingress?

File: interfaces/mimir_worker/v0/README.md [hunk 3]
```diff
@@ -0,0 +1,71 @@
+# `mimir_worker`
+
+## Usage
+
+This relation interface is used by [the Mimir Coordinator charm](/url) and [the Mimir Worker charm](/url) to exchange data.
```
Comment (gruyaume) [inline interfaces/mimir_worker/v0/README.md:5]: Not sure if this phrasing is clearer?
```suggestion
This relation interface is used by [the Mimir Coordinator charm](/url) and [the Mimir Worker charm](/url) to exchange Mimir Configuration data.
```

And I would remove the three bullet points below as they are re-write of the "Behavior" section.

**Review by gruyaume — CHANGES_REQUESTED** [review]

File: docs/json_schemas/mimir_worker/v0/provider.json (hunk 2 above)
Comment (simskij) [inline docs/json_schemas/mimir_worker/v0/provider.json:3]:
```suggestion
  "description": "Provider schema for Mimir Worker.",
```

Comment (simskij) [reply docs/json_schemas/mimir_worker/v0/provider.json:3]: @PietroPasotti can you please regenerate the json schemas from pydantic?

File: interfaces/mimir_worker/v0/README.md [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+- The Mimir Worker charm needs to tell to the coordinator which roles it has
+- The Mimir Coordinator charm needs to tell the worker charms what their configuration is
+- The Mimir Coordinator charm needs to relate to loki and Mimir tell the worker charms what endpoints to push their telemetry to
```
Comment (simskij) [inline interfaces/mimir_worker/v0/README.md:8]:
```suggestion
```

File: interfaces/mimir_worker/v0/README.md [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+```mermaid
+flowchart TD
+    Requirer -- Config,Endpoints --> Provider
+    Provider -- Roles --> Requirer
+```
+
+## Behavior
+### Provider
+
+- Is expected to communicate which roles it was assigned by the cloud admin
```
Comment (simskij) [inline interfaces/mimir_worker/v0/README.md:24]:
```suggestion
- Is expected to communicate which roles it has
```

File: interfaces/mimir_worker/v0/README.md [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+- Is expected to accept any configuration provided by the Coordinator and comply with it as if it came from juju-config 
+- List of expectations that the provider needs to fulfill.
```
Comment (simskij) [inline interfaces/mimir_worker/v0/README.md:26]:
```suggestion
```

File: interfaces/mimir_worker/v0/README.md [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+
+- Is expected to communicate which roles it was assigned by the cloud admin
+- Is expected to accept any configuration provided by the Coordinator and comply with it as if it came from juju-config 
+- List of expectations that the provider needs to fulfill. 
+    
+
+### Requirer
+
+- Is expected to communicate to each worker node:
+  - a mapping from unique identifiers to component roles and URLs
```
Comment (simskij) [inline interfaces/mimir_worker/v0/README.md:32]:
```suggestion
- Is expected to communicate to each worker node:
  - a mapping from unique identifiers to component roles and URLs
  - what their current configuration should be
  - what endpoint(s) to push telemetry to
```

**Review by simskij — CHANGES_REQUESTED** [review]

**Comment by gruyaume** [comment]
@PietroPasotti this PR is pretty stale, any problem if I close it?

**Comment by PietroPasotti** [comment]
fine with me. we'll pick it up when we resume working on mimir next cycle

**Pull request closed without merging**